### PR TITLE
fix: use normal streamhandler for logging

### LINKED ISSUE #1179: v2: Cosmetic issue: verbose logs fill width with spaces
You can see it in action in https://github.com/pdm-project/pdm/issues/1178.
It can be inconvenient when copy/pasting logs from/to various places.
Not sure if it can be changed so that the width is not filled with space.

### LINKED ISSUE #1184: v2 verbose terminal output doesn't take full width
We run `pdm lock -v` in CI and the logs using v2.0.0a have unnecessary newlines. It's very unexpected that a CLI tool limits its output width in this way, and it makes reading through the logs harder.

## expected/before

```text
  Adding requirement googleapis-common-protos<2.0.0dev,>=1.56.0(from googleapis-common-protos 1.56.0)
Conflicts detected: 
  googleapis-common-protos==1.54.0 (from project)
  googleapis-common-protos>=1.5.5 (from <Candidate grpcio-status 1.43.0 from https://pypi.org/simple/grpcio-status/>)
  googleapis-common-protos<2.0dev,>=1.6.0 (from <Candidate google-api-core 1.31.6 from https://pypi.org/simple/google-api-core/>)
  googleapis-common-protos<2.0.0dev,>=1.56.0 (from <Candidate googleapis-common-protos 1.56.3 from https://pypi.org/simple/googleapis-common-protos/>)
  googleapis-common-protos==1.54.0 (from project)
  googleapis-common-protos>=1.5.5 (from <Candidate grpcio-status 1.43.0 from https://pypi.org/simple/grpcio-status/>)
  googleapis-common-protos<2.0dev,>=1.6.0 (from <Candidate google-api-core 1.31.6 from https://pypi.org/simple/google-api-core/>)
```

## actual/now

```text
pdm.termui:   Adding requirement googleapis-common-protos<2.0.0dev,>=1.56.0(from
googleapis-common-protos 1.56.0)                                                
pdm.termui: Conflicts detected:                                                 
  googleapis-common-protos==1.54.0 (from project)                               
  googleapis-common-protos>=1.5.5 (from <Candidate grpcio-status 1.43.0 from htt
ps://gitlab.beta.team/api/v4/projects/beta%2Fpypi/packages/pypi/simple/grpcio-st
atus/>)                                                                         
  googleapis-common-protos<2.0dev,>=1.6.0 (from <Candidate google-api-core      
1.31.6 from https://gitlab.beta.team/api/v4/projects/beta%2Fpypi/packages/pypi/s
imple/google-api-core/>)                                                        
  googleapis-common-protos<2.0.0dev,>=1.56.0 (from <Candidate                   
googleapis-common-protos 1.56.3 from https://gitlab.beta.team/api/v4/projects/be
ta%2Fpypi/packages/pypi/simple/googleapis-common-protos/>)                      
  googleapis-common-protos==1.54.0 (from project)                               
  googleapis-common-protos>=1.5.5 (from <Candidate grpcio-status 1.43.0 from htt
ps://gitlab.beta.team/api/v4/projects/beta%2Fpypi/packages/pypi/simple/grpcio-st
atus/>)                                                                         
  googleapis-common-protos<2.0dev,>=1.6.0 (from <Candidate google-api-core      
1.31.6 from https://gitlab.beta.team/api/v4/projects/beta%2Fpypi/packages/pypi/s
imple/google-api-core/>)                                                        
```

### PR DESCRIPTION
**Pull request by frostming**
## Pull Request Check List

- [x] A news fragment is added in `news/` describing what is new.
- [ ] Test cases added for changed code.

## Describe what you have changed in this PR.
Fix #1179
Fix #1184

### CODE REVIEW

File: pdm/termui.py [hunk 1]
```diff
@@ -216,21 +215,18 @@ def logging(self, type_: str = "install") -> Iterator[logging.Logger]:
...
-            )
+            handler: logging.Handler = logging.StreamHandler()
             handler.setLevel(LOG_LEVELS[self.verbosity])
         else:
             handler = logging.FileHandler(file_name, encoding="utf-8")
             handler.setLevel(logging.DEBUG)
         handler.setFormatter(logging.Formatter("%(name)s: %(message)s"))
-        logger.handlers[1:] = unearch_logger.handlers[:] = [handler]
+        logger.addHandler(handler)
+        unearch_logger.addHandler(handler)
```
Comment (Hnasar) [inline pdm/termui.py:225]: sorry for all the typo fixes, but this should be `unearth` :sweat_smile:

Comment (frostming) [reply pdm/termui.py]: Thanks, fixed. I dunno why I always mistype that word, yeah, not the first time.

File: pdm/termui.py [hunk 2]
```diff
@@ -213,24 +212,23 @@ def logging(self, type_: str = "install") -> Iterator[logging.Logger]:
         """A context manager that opens a file for logging when verbosity is NORMAL or
         print to the stdout otherwise.
         """
-        file_name = mktemp(".log", f"pdm-{type_}-")
-
+        file_name: str | None = None
         if self.verbosity >= Verbosity.DETAIL:
-            handler: logging.Handler = RichHandler(
-                console=_err_console, show_time=False, show_level=False, show_path=False
-            )
```
Comment (Hnasar) [inline pdm/termui.py:221]: I guess this is a bug in rich?
```
        width (int, optional): The width of the terminal. Leave as default to auto-detect width.
```
it seems like it was failing to auto-detect the width when run from docker :raised_eyebrow: 

StreamHandler (which uses stderr) seems safer. :+1: from me. Thanks so much FrostMing!

Comment (frostming) [reply pdm/termui.py:221]: Yes, RichHandler is designed for TTY usage. It wraps or crops text based on the terminal width, and fills the line with spaces(#1179 ). Changing it to a normal stream handler can easily solve it.

**Pull request merged**